components/camel-asterisk/src/main/java/org/apache/camel/component/asterisk/AsteriskConsumer.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.asterisk;
+
+import org.apache.camel.Processor;
+import org.apache.camel.impl.DefaultConsumer;
+
+/**
+ * The Asterisk consumer.
+ */
+public class AsteriskConsumer extends DefaultConsumer {
+    private final AsteriskEndpoint endpoint;
+    private final AsteriskListenerTask task;
+
+    public AsteriskConsumer(AsteriskEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+        this.endpoint = endpoint;
+        this.task = new AsteriskListenerTask(endpoint, this);
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+        log.info("Starting Asterisk AMI Event Listener");
+        endpoint.addListener(task);
+        endpoint.login();
```

Comment:
This could be a problem if you set the Endpoint as singleton, see AsteriskEndpoint::isSingleton